tests/unit/browser/webkit/test_downloads.py

```diff
@@ -30,6 +30,30 @@ def test_download_model(qapp, qtmodeltester, config_stub, cookiejar_and_cache):
     qtmodeltester.check(model)
 
 
+@pytest.mark.parametrize('url, title, out', [
+    ('https://qutebrowser.org/img/cheatsheet-big.png',
+     'cheatsheet-big.png (3342×2060)',
+     None),
+    ('http://qutebrowser.org/INSTALL.html',
+     'Installing qutebrowser | qutebrowser',
+     'Installing qutebrowser _ qutebrowser.html'),
+    ('http://qutebrowser.org/INSTALL.html.html',
+     'Installing qutebrowser | qutebrowser',
+     'Installing qutebrowser _ qutebrowser.html'),
+    ('http://qutebrowser.org/',
+     'qutebrowser | qutebrowser',
+     'qutebrowser _ qutebrowser'),
+    ('https://github.com/qutebrowser/qutebrowser/releases',
+     'Releases · qutebrowser/qutebrowser',
+     'Releases · qutebrowser_qutebrowser'),
+    ('http://qutebrowser.org/page-with-no-title.html',
+     '',
+     None),
+])
```

Comment:
Please also add a test with an upper-case extension (to make sure the title is used), and with an upper-case `.HTML` in the title (see comment above).